src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTaskLogic.cs

```diff
@@ -575,5 +582,22 @@ private void PopulateProjectAndPackageReferences(PackageBuilder packageBuilder,
                 }
             }
         }
+
+        private static IDictionary<string, string> ParsePropertiesAsDictionary(string[] properties)
+        {
+            var dictionary = new Dictionary<string, string>();
+            foreach (var item in properties)
+            {
+                string[] pair = item.Split('=');
+                if (pair.Length != 2)
```

Comment:
Split by `=` and use first token as property name and all the rest portion coming after the first `=` will be the property value with `"` stripped out.

Parsing is fun as it always involves tons of edge cases 😜 .